### DISCUSSION TITLE
[Suggestions] AI personalities, Builder, Explorer and Warrior

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#8817] [Ideas]
Currently, with version 1.1.0, if I have understood correctly, personalities are not active.
Here are some suggestions for different AI player personalities.

AI personalities for all difficulty levels:
Values are given as an example and are subject to adjustment.

- Builder: +1000 gold per day (taken into account in the calculation base for the AI gold bonus for difficulty levels).
Global strategy: more inclined to capture resource generators and piles of resources (gold included).
Castle development strategy: build resource generators (gold) and city defenses as a priority.

- Explorer: +1 move (+100 points) and +1 square viewable area for all heroes.
Global strategy: more inclined to discover the adventure map and to visit the pyramids (when its heroes have the necessary level of wisdom).
Castle development strategy: build creature generators to give its heroes greater mobility on the adventure map as a priority.
Build the highest levels of the mage guilds as a second priority (when its heroes have the necessary level of wisdom).

- Warrior: +100% XP from combat for all heroes.
Global strategy: more inclined to engage in combat (wandering creatures and locations protected by guardians, etc.) and to capture neutral towns, but still with caution, heroes should not become "suicidal".
Castle development strategy: build creature generators as a priority.

**Comment by ihhub** [comment]
My question is why? We must not think from the perspective of cheats for AI but behavior. We will **not** give anymore free stuff for AI.

Also, nobody said that we have to follow the original AI classification.

**Reply by LeHerosInconnu** [reply]
Hello @ihhub,

Okay, we shouldn't give bonuses to the AI then.
Personally, I'm 100% for that. :)
